vm/vm.go

```diff
@@ -157,35 +157,74 @@ func (mod) hex() []uint8 {
 	return []uint8{uint8(opcode.Mod)}
 }
 
-// TODO: implement me w/ test cases :-)
 func (lt) Do(stack *stack, _ asmReader) error {
+	y := stack.pop()
+	x := stack.pop()
+
+	Y := int(y)
+	X := int(x)
+
+	if X < Y {
+		stack.push(item(1))
+	} else {
+		stack.push(item(0))
+	}
+
 	return nil
 }
 
 func (lt) hex() []uint8 {
 	return []uint8{uint8(opcode.LT)}
 }
 
-// TODO: implement me w/ test cases :-)
 func (gt) Do(stack *stack, _ asmReader) error {
+	y := stack.pop()
+	x := stack.pop()
+
+	Y := int(y)
+	X := int(x)
+
+	if X > Y {
+		stack.push(item(1))
+	} else {
+		stack.push(item(0))
+	}
+
 	return nil
 }
 
 func (gt) hex() []uint8 {
 	return []uint8{uint8(opcode.GT)}
 }
 
-// TODO: implement me w/ test cases :-)
 func (eq) Do(stack *stack, _ asmReader) error {
+	y := stack.pop()
+	x := stack.pop()
+
+	Y := int(y)
+	X := int(x)
+
+	if X == Y {
+		stack.push(item(1))
+	} else {
+		stack.push(item(0))
+	}
+
 	return nil
 }
 
 func (eq) hex() []uint8 {
 	return []uint8{uint8(opcode.EQ)}
 }
 
-// TODO: implement me w/ test cases :-)
 func (not) Do(stack *stack, _ asmReader) error {
+	x := stack.pop()
+
+	X := int(x)
+
```

Comment:
Or, Like this, (I think) we can make function which pops just one element is stack, and returns after converting int.

Reply:
I think we can make function which convert `item` to `int` in stack.go
and also i think we should convert `item` to `int32` in above function. because `item` is uint32 and golang recognize the `int` or `uint` types as 32 bits in 32-bit system, 64 bits in 64-bit system. 
the reason why convert to `int32` is that if when we convert `uint32` to `int64` and `uint32` has negative number, `int64` can't recognize the negative number.